store/memstore/credential.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2021 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package memstore
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	errgo "gopkg.in/errgo.v1"
+
+	"github.com/canonical/candid/store"
+)
+
+func mfaCredentialKey(cred store.MFACredential) string {
+	return fmt.Sprintf("%s-%s", cred.ProviderID, cred.Name)
+}
+
+// AddMFACredential stores the specified multi-factor credential.
+func (s *memStore) AddMFACredential(ctx context.Context, cred store.MFACredential) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	key := mfaCredentialKey(cred)
+	_, ok := s.credentials[key]
+	if ok {
+		return errgo.WithCausef(nil, store.ErrDuplicateCredential, "credential with name %q already exists", cred.Name)
+	}
+	s.credentials[key] = cred
+	return nil
+}
+
+// RemoveMFACredential removes the multi-factor credential with the
+// specified username and credential name.
+func (s *memStore) RemoveMFACredential(ctx context.Context, providerID, name string) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	key := fmt.Sprintf("%s-%s", providerID, name)
```

Comment:
why not use your mfaCredentialKey function?

Reply:
done